lalrpop/src/normalize/prevalidate/mod.rs

```diff
@@ -124,6 +124,9 @@ impl<'grammar> Validator<'grammar> {
                     }
                 }
                 GrammarItem::Nonterminal(ref data) => {
+                    if data.public && !data.args.is_empty() {
+                        return_err!(data.span, "macros cannot be marked public");
+                    }
```

Comment:
Can we add a test for this?

Something like this

https://github.com/lalrpop/lalrpop/blob/9bd36041ad3e3cebcabe58b4be7cdd833d4878dd/lalrpop/src/normalize/prevalidate/test.rs#L10-L16

Reply:
Sure. I wasn't sure how to add a negative test, but now I do.